src/eei.cpp

```diff
@@ -45,6 +46,25 @@ bool exceedsUint128(evmc_uint256be const& value) noexcept
 }
 }  // namespace
 
+bool WasmEngine::benchmarkingEnabled = false;
+
+void WasmEngine::collectBenchmarkingData()
+{
+  // Convert duration to string with microsecond units.
+  constexpr auto to_us_str = [](clock::duration d) {
+    return std::to_string(std::chrono::duration_cast<std::chrono::microseconds>(d).count());
+  };
+
+  const auto now = clock::now();
+  const auto instantiationDuration = executionStartTime - instantiationStartTime;
+  const auto executionDuration = now - executionStartTime;
+
+  const auto log = "Time [us]: " + to_us_str(instantiationDuration + executionDuration) + " = " +
+                   to_us_str(instantiationDuration) + " + " + to_us_str(executionDuration) + "\n";
+  std::cerr << log;
+  std::ofstream{"hera_benchmarks.log", std::ios::out | std::ios::app} << log;
```

Comment:
To solve the problem of calls showing the times for children first, perhaps you could pass `msg` to `collectBenchmarkingData` and include some identifiers in the log file, such as `msg.depth`, `msg.sender`, `msg.destination`.

Reply:
This is a good idea. But this is unnecessary for all current benchmarks which either (i) execute one function, or (ii) a bencher function calls the function to be benchmarked, possibly multiple times.